scripts/run-pyright.py

```diff
@@ -313,18 +333,42 @@ def normalize_env(env: str, rebuild: bool, update_pins: bool) -> None:
     return None
 
 
+# Pendulum 2.x has to be built into a wheel. The build config has an undeclared dependency on
+# setuptools that causes build using default behavior, build isolation, to fail. uv does not yet
+# support the `pip install --no-build-isolation` flag, so we separately install pendulum using pip.
+def get_pendulum_install_commands(venv_path: str, src_requirements_path: str) -> Sequence[str]:
+    with open(src_requirements_path, "r") as f:
+        requirements = f.readlines()
+        pendulum_spec = next(
+            (ln.strip() for ln in requirements if ln.startswith("pendulum==")), None
+        )
+    return (
+        [
+            f"{venv_path}/bin/pip install poetry",
+            f"{venv_path}/bin/pip install --no-build-isolation {pendulum_spec}",
+        ]
+        if pendulum_spec
+        else []
+    )
+
+
 def update_pinned_requirements(env: str) -> None:
     print(f"Updating pinned requirements for pyright environment {env}...")
     venv_path = os.path.join(get_env_path(env), ".venv")
+    python_path = f"{venv_path}/bin/python"
     raw_dep_list = subprocess.run(
-        f"{venv_path}/bin/pip freeze", capture_output=True, shell=True, text=True, check=True
+        f"uv pip freeze --python={python_path}",
+        capture_output=True,
+        shell=True,
+        text=True,
+        check=True,
     ).stdout
     is_internal = not os.path.exists("python_modules/dagster")
-    oss_root = "${DAGSTER_GIT_REPO_DIR}/" if is_internal else ""
-    dep_list = re.sub(
-        r"-e git.*?dagster-io/dagster.*?\&subdirectory=(.+)", f"-e {oss_root}\\1", raw_dep_list
-    )
-    dep_list = re.sub(r"-e git.*?dagster-io/internal.*?\&subdirectory=(.+)", "-e \\1", dep_list)
+    oss_root_ref = "${DAGSTER_GIT_REPO_DIR}/" if is_internal else ""
+    resolved_oss_root = os.environ["DAGSTER_GIT_REPO_DIR"].rstrip("/")
+    resolved_internal_root = os.environ["DAGSTER_INTERNAL_GIT_REPO_DIR"].rstrip("/")
+    dep_list = re.sub(f"-e file://{resolved_oss_root}/(.+)", f"-e {oss_root_ref}\\1", raw_dep_list)
+    dep_list = re.sub(f"-e file://{resolved_internal_root}/(.+)", "-e \\1", dep_list)
```

Comment:
The output format of `uv pip freeze` is slightly different than `pip freeze` so has to be differently massaged.